tgui/packages/tgui/interfaces/CircuitSignalHandler.tsx

```diff
@@ -0,0 +1,249 @@
+import { Component } from "inferno";
+import { useBackend } from "../backend";
+import { Box, Stack, Section, Input, Button, Dropdown } from "../components";
+import { Window } from "../layouts";
+
+type Response = {
+  name: string;
+  bitflag: number;
+}
+
+type Parameter = {
+  name: string;
+  datatype: string;
+}
+
+type CircuitSignalHandlerState = {
+  signal_id: string;
+  responseList: Response[];
+  parameterList: Parameter[];
+  global: Boolean;
+}
+
+type CircuitSignalHandlerData ={
+  global_port_types: string[];
+}
+
+type BitflagToString = {
+  [key: number]: string;
+}
+
+export class CircuitSignalHandler
+  extends Component<{}, CircuitSignalHandlerState> {
+  bitflags: BitflagToString;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      signal_id: "signal_id",
+      responseList: [],
+      parameterList: [],
+      global: false,
+    };
+
+    this.bitflags = {};
+
+    for (let i = 0; i < 24; i++) {
+      this.bitflags[1 << i] = `Flag ${i+1}`;
+    }
+  }
+
+  render() {
+    const { act, data } = useBackend<CircuitSignalHandlerData>(this.context);
+    const { responseList, parameterList, signal_id, global }
+      = this.state as CircuitSignalHandlerState;
+    const {
+      global_port_types,
+    } = data;
+    return (
+      <Window width={600} height={300}>
+        <Window.Content>
+          <Stack vertical fill>
+            <Stack.Item>
+              <Stack fill>
+                <Stack.Item grow>
+                  <Input
+                    placeholder="Signal ID"
+                    value={signal_id}
+                    fluid
+                    onChange={(e, value) => this.setState({ signal_id: value })}
+                  />
+                </Stack.Item>
+                <Stack.Item>
+                  <Button.Checkbox
+                    checked={global}
+                    content="Global"
+                    onClick={(e) => this.setState({ global: !global })}
+                  />
+                </Stack.Item>
+              </Stack>
+            </Stack.Item>
+            <Stack.Item grow>
+              <Stack width="100%" height="100%">
+                <Stack.Item basis="50%">
```

Comment:
Each of the items should use flexible units `grow={1} basis={0}` instead of percentage units `basis="50%"`